API/Controllers/AccountController.cs

```diff
@@ -508,6 +508,19 @@ public async Task<ActionResult> UpdateAccount(UpdateUserDto dto)
             await _userManager.UpdateNormalizedUserNameAsync(user);
             _unitOfWork.UserRepository.Update(user);
         }
+    
+        // Update email
+        if (!string.IsNullOrEmpty(dto.Email) && !user.Email!.Equals(dto.Email))
+        {
+            // Validate email change
+            var emailValidationErrors = await _accountService.ValidateEmail(dto.Email);
+            if (emailValidationErrors.Any()) return BadRequest(await _localizationService.Translate(User.GetUserId(), "email-taken"));
+            user.Email = dto.Email;
+            user.EmailConfirmed = false;
+            user.NormalizedEmail = _userManager.NormalizeEmail(dto.Email);
+            user.ConfirmationToken = await _userManager.GenerateEmailConfirmationTokenAsync(user);
```

Comment:
You generate a confirmation token, but I don't see you triggering an email for the end user to confirm and get rid of that pending status.